### DISCUSSION TITLE
Library "ppx_dhall_type" not found

### DISCUSSION THREAD

**Discussion by GavinXu520** [MinaProtocol/mina#14448] [Q&A]
When I executed a dune build in the root directory of mina, I reported an error: Library "ppx_dhall_type" not found. I found out that this is a local library, but how can I obtain this local library?
The error is as follows:
![image](https://github.com/MinaProtocol/mina/assets/13655605/d775a342-dafe-4a7b-9cb0-ae551583b1d5)
Through a global search for 'ppx_dhall_type', it was found that the dune file in the directory 'mini/src/app/hall_types' has a reference to' ppx_dhall_type 'and is declared as a' local library ':
![1698652009594](https://github.com/MinaProtocol/mina/assets/13655605/4e003869-e492-47dd-8d6c-752fef49b3f6)
And used in 'mini/src/app/hall_types/dump_dhall_types. ml':
![1698652150092](https://github.com/MinaProtocol/mina/assets/13655605/f5223666-75c2-4305-8123-aa7574592384)
So how can I obtain this' local library '?
Is there any expert who knows and can provide guidance? Thank you very much!

**Comment by shimkiv** [comment]
Hey @GavinXu520 can you describe your build procedure in more details please?

https://github.com/MinaProtocol/mina/blob/develop/README-dev.md#building

> Some dependencies that are not taken from opam or integrated with dune must be added manually. Run the `scripts/pin-external-packages.sh` script.
